File: app/src/main/java/com/varunbarad/popularmovies/model/data/MovieStub.java
package com.varunbarad.popularmovies.model.data;

import com.google.gson.Gson;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

/**
 * Creator: vbarad
 * Date: 2016-11-24
 * Project: PopularMovies
 */

public class MovieStub {
  @SerializedName("poster_path")
  @Expose
  private String posterPath;
  @SerializedName("adult")
  @Expose
  private boolean adult;
  @SerializedName("overview")
  @Expose
  private String overview;
  @SerializedName("release_date")
  @Expose
  private String releaseDate;
  @SerializedName("genre_ids")
  @Expose
  private ArrayList<Long> genreIds;
  @SerializedName("id")
  @Expose
  private long id;
  @SerializedName("original_title")
  @Expose
  private String originalTitle;
  @SerializedName("original_language")
  @Expose
  private String originalLanguage;
  @SerializedName("title")
  @Expose
  private String title;
  @SerializedName("backdrop_path")
  @Expose
  private String backdropPath;
  @SerializedName("popularity")
  @Expose
  private double popularity;
  @SerializedName("vote_count")
  @Expose
  private long voteCount;
  @SerializedName("video")
  @Expose
  private boolean video;
  @SerializedName("vote_average")
  @Expose
  private double voteAverage;

  /**
   * No args constructor for use in serialization
   */
  public MovieStub() {
  }

  /**
   * @param id
   * @param genreIds
   * @param title
   * @param releaseDate
   * @param overview
   * @param posterPath
   * @param originalTitle
   * @param voteAverage
   * @param originalLanguage
   * @param adult
   * @param backdropPath
   * @param voteCount
   * @param video
   * @param popularity
   */
  public MovieStub(String posterPath, boolean adult, String overview, String releaseDate, ArrayList<Long> genreIds, long id, String originalTitle, String originalLanguage, String title, String backdropPath, double popularity, long voteCount, boolean video, double voteAverage) {
    this.posterPath = posterPath;
    this.adult = adult;
    this.overview = overview;
    this.releaseDate = releaseDate;
    this.genreIds = genreIds;
    this.id = id;
    this.originalTitle = originalTitle;
    this.originalLanguage = originalLanguage;
    this.title = title;
    this.backdropPath = backdropPath;
    this.popularity = popularity;
    this.voteCount = voteCount;
    this.video = video;
    this.voteAverage = voteAverage;
  }

  /**
   * @return The image path for poster
   */
  public String getPosterPath() {
    return posterPath;
  }

  /**
   * @return `true` if the movie is an adult movie
   */
  public boolean isAdult() {
    return adult;
  }

  /**
   * @return The overview
   */
  public String getOverview() {
    return overview;
  }

  /**
   * @return The releaseDate
   */
  public String getReleaseDate() {
    return releaseDate;
  }

  /**
   * @return The genreIds
   */
  public ArrayList<Long> getGenreIds() {
    return genreIds;
  }

  /**
   * @return The id
   */
  public long getId() {
    return id;
  }

  /**
   * @return The originalTitle
   */
  public String getOriginalTitle() {
    return originalTitle;
  }

  /**
   * @return The originalLanguage
   */
  public String getOriginalLanguage() {
    return originalLanguage;
  }

  /**
   * @return The title
   */
  public String getTitle() {
    return title;
  }

  /**
   * @return The backdropPath
   */
  public String getBackdropPath() {
    return backdropPath;
  }

  /**
   * @return The popularity
   */
  public double getPopularity() {
    return popularity;
  }

  /**
   * @return The voteCount
   */
  public long getVoteCount() {
    return voteCount;
  }

  /**
   * @return The video
   */
  public boolean isVideo() {
    return video;
  }

  /**
   * @return The voteAverage
   */
  public double getVoteAverage() {
    return voteAverage;
  }

  @Override
  public String toString() {
    return (new Gson()).toJson(this);
  }
}

File: app/src/main/java/com/varunbarad/popularmovies/model/data/MovieStub.kt
package com.varunbarad.popularmovies.model.data

import android.os.Parcel
import android.os.Parcelable
import com.varunbarad.popularmovies.external_services.movie_db_api.models.ApiMovieStub

/**
 * Creator: Varun Barad
 * Date: 2019-06-05
 * Project: PopularMovies
 */
data class MovieStub(
    val posterPath: String,
    val adult: Boolean = false,
    val overview: String = "",
    val releaseDate: String? = null,
    val id: Long,
    val title: String,
    val backdropPath: String = "",
    val voteAverage: Double
) : Parcelable {
    constructor(parcel: Parcel) : this(
        (parcel.readString() ?: ""),
        (parcel.readByte() != 0.toByte()),
        (parcel.readString() ?: ""),
        parcel.readString(),
        parcel.readLong(),
        (parcel.readString() ?: ""),
        (parcel.readString() ?: ""),
        parcel.readDouble()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(posterPath)
        parcel.writeByte(if (adult) 1 else 0)
        parcel.writeString(overview)
        parcel.writeString(releaseDate)
        parcel.writeLong(id)
        parcel.writeString(title)
        parcel.writeString(backdropPath)
        parcel.writeDouble(voteAverage)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<MovieStub> {
        override fun createFromParcel(parcel: Parcel): MovieStub {
            return MovieStub(parcel)
        }

        override fun newArray(size: Int): Array<MovieStub?> {
            return arrayOfNulls(size)
        }
    }
}

fun ApiMovieStub.toMovieStub(): MovieStub {
    return MovieStub(
        posterPath = this.posterPath,
        adult = this.adult,
        overview = this.overview,
        releaseDate = this.releaseDate,
        id = this.id,
        title = this.title,
        backdropPath = this.backdropPath,
        voteAverage = this.voteAverage
    )
}
